backend-api/src/functions/common/utils/environment.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  errorResult,
+  Result,
+  Success,
+  successResult,
+} from "../../utils/result";
+
+export type MissingEnvVarError = {
+  missingEnvVars: string[];
+};
+
+export type Config<T extends string> = {
+  [key in T]: string;
+};
+
+export const getRequiredEnvironmentVariables = <T extends string>(
```

Comment:
Made another comment re envVars, would be good to chat about the general approach here. Looking at this func it feels a bit weird but unsure if it's just because of the generics and typings of stuff look a bit different to me. Not against at all but yeah, would be useful for me to chat through

Reply:
Happy to chat through it any time (and see if it can be improved, too!). The goal was to have a generic function that could take a list of strings for the env vars we want to retrieve, and give back either a type-safe map with all these strings as keys, or an error. This way, the config functions for the individual lambdas only need to specify once which env vars they need. The pursuit of type-safety is why we've had to use all the generics etc - but there may well be a simpler way to do it.